include/cling/Interpreter/Interpreter.h

```diff
@@ -195,6 +196,12 @@ namespace cling {
     ///
     mutable std::vector<ClangInternalState*> m_StoredStates;
 
+    enum {
+      kStdStringTransaction = 0, // Transaction known to contain std::string
+      kNumTransactions
+    };
+    mutable const Transaction* m_CachedTrns[kNumTransactions];
```

Comment:
The next PR will expand this to 2 elements.
There are also problems when rolling back to the state before **cling::printValue** was invoked for the first time that can be solved using the same mechanism.